aten/src/ATen/native/cpu/QuickPartition.cpp

```diff
@@ -0,0 +1,556 @@
+#include <aten/src/ATen/native/cpu/QuickPartition.h>
+#include <stdint.h>
+#if CPU_CAPABILITY_AVX2
+#include <aten/src/ATen/cpu/vec256/vec256.h>
+#include <xmmintrin.h>
+#include <immintrin.h>
+#include <algorithm>
+#include <limits>
+#include <cmath>
+
+using at::vec256::Vec256;
+using at::vec256::permute;
+using at::vec256::int_same_size_t;
+
+
+template <typename T>
+struct Table;
+
+template <>
+struct Table<float> {
+  static const uint32_t __attribute__((aligned(256))) t[256*8];
+};
+
+const uint32_t __attribute__((aligned(256))) Table<float>::t[] = {
```

Comment:
A comment explaining how this has been computed and what it is would be very useful

Reply:
Good point, I have added a comment

Reply:
The comment says what the table is (good!) but it doesn't say how to compute the table (you probably ran a Python script or something to generate the table, right? Paste it in here.)

Reply:
I'd rather see something like this in vec256 directly, since it's very specific to 256bit vectors.